### ISSUE TITLE
Error 400: Invalid IdP when trying o_auth_login

### ISSUE THREAD

**Issue by Bmoser-boop** [Kohze/fireData#23]
Hi,
I am attempting to us the function o_auth_login but keep getting an error saying that my IdP is invalid. My process is as follows:
1. I start with auth() to create login_token, and this works successfully.
     login_token <- auth(projectAPI = “API”, email = “email”, password = “password”)
2. I attempt o_auth_login with the following information:
     project_api: The same as I used in auth()
     request_uri: the url listed under project_info.firebase_url in the JSON file of the site.
     post_body = login_token$idToken
     return_idp_credential=T
When I attempt this a window opens with the following error messsage:
**Error 400: redirect_uri_mismatch**
The redirect URI in the request, http://localhost:1410/, does not match the ones authorized for the OAtuh client. To update the authorized redirect URIs, visit: (a google apis URL.)

Thanks so much for any help you can give!

**Comment by Kohze** [comment]
Just looked at the function, which itself is quite lightweight. Lets first check if its related to the code or to your credentials.
Try to login you via a "postman" https://www.postman.com/ request, looking at our code, it seems that its just a simple API call that seems to fail here. 

`  o_auth_login_url <- paste0("https://www.googleapis.com/identitytoolkit/v3/relyingparty/verifyAssertion?key=", project_api)
  o_auth_login_data <- httr::POST(url = o_auth_login_url, body = list("requestUri" = request_uri, "postBody" = post_body, 
`

In case the postman API call works, but the package not, we do some bug fixing round.

**Comment by Bmoser-boop** [comment]
Thanks a lot for the quick response! I’ll try to use postman (I haven’t used it before so this may take a little while to figure out) and see if I get better results there.

**Comment by Bmoser-boop** [comment]
Hi,

I just making an API call on Postman with this url:

paste0("https://www.googleapis.com/identitytoolkit/v3/relyingparty/verifyAssertion?key= <https://www.googleapis.com/identitytoolkit/v3/relyingparty/verifyAssertion?key=>", project_api)

And I again received Error 404. Unfortunately I am not familiar with Postman, so was that what I needed to do to check the API call or do I also need to send credentials along with the request?

Thank you very much for your help,

Brian

**Comment by Bmoser-boop** [comment]
As an update, I tried making a separate test Firebase project and tried using this project’s API Key and Database URL and I am receiving the same errors, I don’t know if this changes what the possible error is.

**Comment by paulsp94** [comment]
> The redirect URI in the request, http://localhost:1410/, does not match the ones authorized for the OAuth client. 

The error means that you didn't register the `http://localhost:1410/` URL as a valid redirect option, which is a security feature of OAuth providers.

One example, how to add a URL to e.g. Facebook can be found here:

https://stackoverflow.com/a/40174229

**Comment by Bmoser-boop** [comment]
I was able to get it to work, my partner who is setting up our Firebase project did not create a Real Time Database for our Firebase project. We realized this was the problem after I found that I was able to add and remove users with no issue, but couldn't upload or download any files. I had failed to see that this package works best with the Real Time Database.

Additionally, I realized that an OAuth login was not necessary for what I was trying to do, since with the security settings on my project I could upload and download files without an OAuth login.

As a warning to anybody who is a beginner with Firebase and has this issue, I tried adding http://localhost:1410/ as a valid OAuth link but still received the same errors. If you're trying to upload/download files and getting similar errors as me, consider that the problem might be on the Firebase side rather than the R side.
